sycl/source/backend.cpp

```diff
@@ -202,14 +202,21 @@ make_kernel_bundle(pi_native_handle NativeHandle, const context &TargetContext,
   return make_kernel_bundle(NativeHandle, TargetContext, false, State, Backend);
 }
 
-kernel make_kernel(pi_native_handle NativeHandle, const context &TargetContext,
+kernel make_kernel(const context &TargetContext,
+                   const kernel_bundle<bundle_state::executable> &KernelBundle,
+                   pi_native_handle NativeHandle, bool KeepOwnership,
                    backend Backend) {
   const auto &Plugin = getPlugin(Backend);
   const auto &ContextImpl = getSyclObjImpl(TargetContext);
+  const auto &DeviceImage = KernelBundle.begin();
+  const auto &DeviceImageImpl = getSyclObjImpl(DeviceImage);
```

Comment:
@alexbatashev : I can't see how to properly get PI program from a kernel bundle (we should assert there is only one).

Reply:
I was missing that KernelBundle.begin() is an iterator.